Development

jasonquinto opened the pull request:
Removed deprecated variables and added new variables with fallbacks in the `martini-upload-package` SSM and variable modules to support:
- parameter_name
- package_name_pattern
- package_dir
- async_upload
- success_check_timeou
- success_check_delay
- success_check_package_name

Updated all README files to:
- Explain how to configure and deploy each module
- Document the new inputs and expected SSM parameter structure

jasonquinto committed:
Updated SSM and variable module to reflect changes from upload package buildspec

jasonquinto committed:
Fixed formatting

jasonquinto committed:
Add allowed packages in SSM

jasonquinto committed:
Merge branch 'main' into development

jasonquinto committed:
Applied changes from martini upload pipeline update

jasonquinto committed:
Added parameter defaults

jasonquinto committed:
Updated the README files to reflect the changes

rafaljanicki commented on martini-upload-package/README.md:
```diff
@@ -4,74 +4,72 @@ This Terraform module configures an AWS CodePipeline for uploading Martini packa
-**Note**: CodeStar Connections - The connection must be created manually in the AWS Management Console due to the required authentication flow with the external platform. Once created, it generates a Connection ARN that can be referenced in automation tools like Terraform.
+### Configure Inputs
+
+Update the `terraform.tfvars` file or override variables via CLI to configure:
```
I wonder, maybe we can use official tfdoc over here? This is the table that's usually seen on modules' pages

jasonquinto replied:
Thank you for introducing such a useful plugin! I’ve reworked the inputs table using tfdocs and also corrected the bool type based on your comment below.

rafaljanicki commented on martini-upload-package/variable.tf:
```diff
@@ -58,8 +60,38 @@ variable "martini_access_token" {
+variable "async_upload" {
+  description = "Set to true to treat HTTP 504 as successful upload"
+  type        = string
+  default     = "false"
```
Shouldn't that be a bool?

rafaljanicki replied:
That's still not a bool

jasonquinto committed:
Reworked the inputs reference using tfdocs

jasonquinto committed:
Minor corrections in README

jasonquinto committed:
Removed markeddown links for inputs

rafaljanicki commented on martini-build-image/README.md:
```diff
@@ -72,8 +58,28 @@ Run the following Terraform commands:
+
+| Name                   | Description                                                           | Type         | Default              | Required |
+|------------------------|-----------------------------------------------------------------------|--------------|----------------------|:--------:|
+| aws_account_id         | AWS Account ID                                                       | string       | n/a                  | yes      |
```
That doesn't need to be an input - https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/caller_identity.html

rafaljanicki commented on martini-build-image/README.md:
```diff
@@ -72,8 +58,28 @@ Run the following Terraform commands:
+| Name                   | Description                                                           | Type         | Default              | Required |
+|------------------------|-----------------------------------------------------------------------|--------------|----------------------|:--------:|
+| aws_account_id         | AWS Account ID                                                       | string       | n/a                  | yes      |
+| aws_region             | AWS region for deployment                                             | string       | n/a                  | yes      |
```
That doesn't need to be an input - https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/region

rafaljanicki commented on martini-build-image/README.md:
```diff
@@ -72,8 +58,28 @@ Run the following Terraform commands:
+| ecr_repo_name          | ECR repository name for Martini builds                                | string       | n/a                  | yes      |
+| environment            | Environment name (e.g., 'dev', 'prod')                                | string       | "dev"                | no       |
+| log_retention_days     | Retention period for CloudWatch logs (in days)                        | number       | n/a                  | yes      |
+| martini_version        | Version of Martini to be used in the build                            | string       | n/a                  | yes      |
```
Can we default to `latest`?

rafaljanicki commented on martini-upload-package/README.md:
```diff
@@ -4,74 +4,71 @@ This Terraform module configures an AWS CodePipeline for uploading Martini packa
+| Name                         | Description                                                               | Type         | Default                  | Required |
+|------------------------------|---------------------------------------------------------------------------|--------------|--------------------------|:--------:|
+| async_upload                 | Set to true to treat HTTP 504 as successful upload                        | bool         | "false"                  | no       |
+| aws_region                   | AWS region for deployment                                                 | string       | n/a                      | yes      |
```
See above

rafaljanicki commented on martini-upload-package/README.md:
```diff
@@ -4,74 +4,71 @@ This Terraform module configures an AWS CodePipeline for uploading Martini packa
+
+| Name                         | Description                                                               | Type         | Default                  | Required |
+|------------------------------|---------------------------------------------------------------------------|--------------|--------------------------|:--------:|
+| async_upload                 | Set to true to treat HTTP 504 as successful upload                        | bool         | "false"                  | no       |
```
"false" is not a bool

jasonquinto committed:
Removed AWS Region and ID from inputs

jasonquinto committed:
Merge branch 'main' into development

jasonquinto commented:
@rafaljanicki I’ve removed the need for AWS account ID and region from the inputs and fixed the bool type issue. For `martini_version`, Docker already defaults to latest when no tag is provided, but to make the behavior clearer and more intentional, I explicitly set it to default to latest when no input is specified.
